cmd/integration_test.go

```diff
@@ -306,85 +328,109 @@ func TestSmokescreenIntegration(t *testing.T) {
-						ExpectAllow: authorizedHost || action != acl.Enforce,
-						OverTls:     overTls,
-						OverConnect: overConnect,
-						ProxyURL:    servers[overTls].URL,
-						TargetPort:  outsideListenerPort,
-						Host:        host,
-						RoleName:    generateRoleForAction(action),
+						ExpectAllow:  expectAllow,
+						Action:       action,
+						OverTls:      overTls,
+						OverConnect:  overConnect,
+						ProxyURL:     servers[overTls].URL,
+						TargetPort:   port,
+						TargetHost:   host,
+						TargetScheme: targetScheme,
+						RoleName:     generateRoleForAction(action),
 					}
 					testCases = append(testCases, testCase)
 				}
 			}
 		}
 
 		baseCase := TestCase{
-			OverConnect: overConnect,
-			ProxyURL:    servers[false].URL,
-			TargetPort:  outsideListenerPort,
+			OverConnect:  overConnect,
+			ProxyURL:     servers[false].URL,
+			TargetPort:   outsideListenerPort,
+			TargetScheme: "http",
 		}
 
+		// Empty roles should default deny per the test config
 		noRoleDenyCase := baseCase
-		noRoleDenyCase.Host = "127.0.0.1"
+		noRoleDenyCase.TargetHost = "127.0.0.1"
 		noRoleDenyCase.ExpectAllow = false
 
-		noRoleAllowCase := baseCase
-		noRoleAllowCase.Host = "localhost"
-		noRoleAllowCase.ExpectAllow = true
-
-		unknownRoleDenyCase := noRoleDenyCase
+		// Unknown roles should default deny per the test config
+		unknownRoleDenyCase := baseCase
+		unknownRoleDenyCase.TargetHost = "127.0.0.1"
 		unknownRoleDenyCase.RoleName = "unknown"
+		unknownRoleDenyCase.ExpectAllow = false
 
-		unknownRoleAllowCase := noRoleAllowCase
-		unknownRoleAllowCase.RoleName = "unknown"
-
-		badIPRangeCase := baseCase
 		// This must be a global unicast, non-loopback address or other IP rules will
 		// block it regardless of the specific configuration we're trying to test.
-		badIPRangeCase.Host = "1.1.1.1"
+		badIPRangeCase := baseCase
+		badIPRangeCase.TargetHost = "1.1.1.1"
 		badIPRangeCase.ExpectAllow = false
 		badIPRangeCase.RoleName = generateRoleForAction(acl.Open)
 
-		badIPAddressCase := baseCase
 		// This must be a global unicast, non-loopback address or other IP rules will
 		// block it regardless of the specific configuration we're trying to test.
-		badIPAddressCase.Host = "1.0.0.1"
+		badIPAddressCase := baseCase
+		badIPAddressCase.TargetHost = "1.0.0.1"
 		badIPAddressCase.TargetPort = 123
 		badIPAddressCase.ExpectAllow = false
 		badIPAddressCase.RoleName = generateRoleForAction(acl.Open)
 
-		proxyCase := baseCase
 		// We expect this URL to always return a non-200 status code so that
 		// this test will fail if we're not respecting the UpstreamProxy setting
 		// and instead going straight to this host.
-		proxyCase.Host = "aws.s3.amazonaws.com"
-		proxyCase.UpstreamProxy = outsideListenerUrl.String()
-		proxyCase.ExpectAllow = true
-		proxyCase.RoleName = generateRoleForAction(acl.Open)
+		if overConnect {
+			proxyCase := baseCase
+			proxyCase.TargetScheme = "https"
+			proxyCase.TargetHost = "aws.s3.amazonaws.com"
```

Comment:
What if https://aws.s3.amazonaws.com/ starts returning 200 or 301->200 or something? IMO it would be better to stand up another temporary HTTP server (as part of test setup) that returns 403 or similar.

Reply:
I removed this test case, but I'm generally in favor of keeping reasonably stable external domains as this is an integration test suite which should be tested against non-local HTTP servers.